services/m3dbnode/config/config.go

```diff
@@ -190,3 +205,146 @@ type HashingConfiguration struct {
 	// Murmur32 seed value.
 	Seed uint32 `yaml:"seed"`
 }
+
+// ETCDConfig creates a new embedded etcd config from kv config.
+func ETCDConfig(cfg Configuration) (*embed.Config, error) {
```

Comment:
minor nit: Perhaps a better name would be `NewEtcdEmbedConfiguration`?  Also this could be instead a member method on `Configuration` itself which would be more similar to our other sub-component creations off of `Configuration`.

e.g.